File: 백준/Bronze/5585. 거스름돈/거스름돈.js
let fs = require("fs");

let N = fs.readFileSync("/dev/stdin").toString().trim().split("\n").map(Number)[0];

let remain = 1000 - N;
let cnt = 0;

// 500 100 50 10 5 1
while(remain>0){
  if(remain>=500){
    cnt += parseInt(remain/500);
    remain -= parseInt(remain/500)*500;
  }else if(remain >=100){
     cnt += parseInt(remain/100);
    remain -= parseInt(remain/100)*100;
  }else if(remain >=50){
     cnt += parseInt(remain/50);
    remain -= parseInt(remain/50)*50;
  }else if(remain >=10){
     cnt += parseInt(remain/10);
    remain -= parseInt(remain/10)*10;
  }else if(remain >=5){
     cnt += parseInt(remain/5);
    remain -= parseInt(remain/5)*5;
  }else{
    cnt += parseInt(remain/1);
    remain -= parseInt(remain/1)*1;
  }
}

console.log(cnt)

File: 백준/Bronze/5585. 거스름돈/거스름돈.java
import java.io.*;
//import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
//        StringTokenizer st;

        int remain = 1000 - Integer.parseInt(br.readLine());
        int[] money = {500, 100, 50, 10, 5, 1};


        int cnt = 0, idx = 0;
        while (remain > 0) {
            cnt += remain / money[idx];
            remain %= money[idx];
            idx++;
        }

        bw.write(String.valueOf(cnt));
        bw.flush();
        bw.close();
        br.close();
    }
}